pytorch_lightning/plugins/precision/double.py

```diff
@@ -90,3 +91,14 @@ def connect(
     def post_dispatch(self) -> None:
         while len(self.patches) > 0:
             self.patches.pop().teardown()
+
+    @contextlib.contextmanager
+    def train_step_context(self) -> Generator:
+        """A contextmanager for the trainstep"""
+        torch.set_default_tensor_type(torch.DoubleTensor)
+        yield
+        torch.set_default_tensor_type(torch.FloatTensor)
```

Comment:
shall you rather save what was it before and then set it back?

Reply:
I think it should be fine. Didn't find any utils to get the default either.